.github/workflows/test-environment.yml

```diff
@@ -0,0 +1,224 @@
+name: Sanity tests
+
+on:
+  # Ability to execute on demand
+  workflow_dispatch:
+    inputs:
+      deployment_name:
+        type: string
+        description: |
+          Name your environment (Only a-zA-Z0-9 and `-`).
+          For example: john-8-7-2-June01'
+        required: true
+      ec-api-key:
+        type: string
+        description: "Elastic Cloud API KEY"
+        required: true
+      elk-stack-version:
+        required: true
+        description: "Elastic Cloud stack SNAPSHOT or BC version"
+        default: "8.7.2-SNAPSHOT"
+      ess-region:
+        required: true
+        description: "Elastic Cloud deployment region"
+        default: "gcp-us-west2"
+      docker-image-override:
+        required: false
+        description: "Provide the full Docker image path to override the default image (BC versions only)"
+      cleanup-env:
+        description: "Cleanup resources after provision"
+        default: false
+        type: boolean
+
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: "eu-west-1"
+  TF_VAR_ec_api_key: ${{ secrets.CI_EC_API_KEY }}
+  WORKING_DIR: deploy/test-environments
+  FLEET_API_DIR: fleet_api/src
+
+jobs:
+  Deploy:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 120
+    defaults:
+      run:
+        working-directory: ${{ env.WORKING_DIR }}
+    env:
+      TF_VAR_stack_version: ${{ github.event.inputs.elk-stack-version }}
+      TF_VAR_ess_region: ${{ github.event.inputs.ess-region }}
+      TF_VAR_ec_api_key: ${{ github.event.inputs.ec-api-key }}
+      DEPLOYMENT_NAME: ${{ github.event.inputs.deployment_name }}
+      S3_BASE_BUCKET: "s3://tf-state-bucket-test-infra"
+      DOCKER_IMAGE_OVERRIDE: ${{ github.event.inputs.docker-image-override }}
+      STACK_VERSION: ${{ github.event.inputs.elk-stack-version }}
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Init Hermit
+        run: ./bin/hermit env -r >> $GITHUB_ENV
+        working-directory: ./
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+
+      - name: Install Poetry
```

Comment:
Add it to hermit instead?

Reply:
I couldn't find a suitable Hermit package for Poetry. However, the [Poetry documentation](https://python-poetry.org/docs/#ci-recommendations) suggests using the official installer.